src/edu/washington/escience/myria/io/ByteSink.java

```diff
@@ -0,0 +1,24 @@
+/**
+ *
+ */
+package edu.washington.escience.myria.io;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class ByteSink implements DataSink {
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  final ByteArrayOutputStream writerOutput;
+
+  public ByteSink() {
+    writerOutput = new ByteArrayOutputStream();
```

Comment:
Is there a good reason not to materialize the `OutputStream` in `getOutputStream()`, just like all the `DataSource` implementations?